staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -185,48 +188,52 @@ func DefaultVerifyOptions() x509.VerifyOptions {
 
 // CommonNameUserConversion builds user info from a certificate chain using the subject's CommonName
 var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (*authenticator.Response, bool, error) {
-	if len(chain[0].Subject.CommonName) == 0 {
-		return nil, false, nil
+	userInfo, succeeded, err := convertToUserInfo(chain[0].Subject)
+	if !succeeded {
+		return nil, false, err
 	}
+	return &authenticator.Response{
+		User: userInfo,
+	}, true, err
+})
 
-	isNewformat := false
-	tenant := ""
-	if len(chain[0].Subject.Organization) > 0 {
-		parts := strings.Split(chain[0].Subject.Organization[0], ":")
-		if len(parts) > 0 && parts[0] == "tenant" {
-			isNewformat = true
-			tenant = parts[1]
-		}
+func convertToUserInfo(subject pkix.Name) (*user.DefaultInfo, bool, error) {
+	errTenantExtraction := errors.New(ErrTenantExtraction)
+	if len(subject.CommonName) == 0 {
+		return nil, false, errTenantExtraction
```

Comment:
Is errors.New("CommonName can not be empty") more descriptive?

Reply:
I'll split the error message for each specific cases.

Reply:
fixed